### PR TITLE
Handle out of scope dances

### LINKED ISSUE #341: "Cannot choose from an empty sequence" error for "jump jump jump"
## Type of Issue

Select the type of issue:
- [x] Bug report (to report a bug)
- [ ] Feature request (to request an additional feature)
- [ ] Tracker (I am just using this as a tracker)
- [ ] Refactor request
- [ ] Documentation Ask

## Description

Thought this error was fixed?

```
2021-04-20 20:21:09,067 [nsp_dialogue_manager.py:172 - handle_logical_form() INFO]: logical form post-coref "d9a8ef880c5e346f7e01c8fee57ad5380d135fb37f98fccd6b3d4d51fbab37ca:b786faa38b9a47ca96c3608963dacea1" -> {'dialogue_type': 'HUMAN_GIVE_COMMAND', 'action_sequence': [{'action_type': 'DANCE', 'dance_type': {'filters': {'triples': [{'pred_text': 'has_name', 'obj_text': 'jump jump'}]}}}]}
ERROR:root:Default handler caught exception, db_log_idx=46625
Traceback (most recent call last):
  File "/private/home/rebeccaqian/droidlet/base_agent/core.py", line 18, in start
    self.step()
  File "/private/home/rebeccaqian/droidlet/base_agent/loco_mc_agent.py", line 216, in step
    super().step()
  File "/private/home/rebeccaqian/droidlet/base_agent/core.py", line 26, in step
    self.controller_step()
  File "/private/home/rebeccaqian/droidlet/base_agent/loco_mc_agent.py", line 282, in controller_step
    self.dialogue_manager.step(incoming_chats[0])
  File "/private/home/rebeccaqian/droidlet/base_agent/dialogue_manager.py", line 100, in step
    self.dialogue_stack.step()
  File "/private/home/rebeccaqian/droidlet/base_agent/dialogue_stack.py", line 54, in step
    output_chat, step_data = self.stack[-1].step()
  File "/private/home/rebeccaqian/droidlet/base_agent/dialogue_objects/interpreter.py", line 103, in step
    r = self.action_handlers[action_type](self.speaker, action_def)
  File "/private/home/rebeccaqian/droidlet/craftassist/agent/dialogue_objects/mc_interpreter.py", line 520, in handle_dance
    return new_tasks(), None, None
  File "/private/home/rebeccaqian/droidlet/craftassist/agent/dialogue_objects/mc_interpreter.py", line 493, in new_tasks
    dance_memid = random.choice(dance_memids)
  File "/private/home/rebeccaqian/.conda/envs/droidlet_env_clone/lib/python3.7/random.py", line 261, in choice
    raise IndexError('Cannot choose from an empty sequence') from None
IndexError: Cannot choose from an empty sequence
2021-04-20 20:21:09,069 [loco_mc_agent.py:192 - handle_exception() ERROR]: Default handler caught exception, db_log_idx=46625
Traceback (most recent call last):
  File "/private/home/rebeccaqian/droidlet/base_agent/core.py", line 18, in start
    self.step()
  File "/private/home/rebeccaqian/droidlet/base_agent/loco_mc_agent.py", line 216, in step
    super().step()
  File "/private/home/rebeccaqian/droidlet/base_agent/core.py", line 26, in step
    self.controller_step()
  File "/private/home/rebeccaqian/droidlet/base_agent/loco_mc_agent.py", line 282, in controller_step
    self.dialogue_manager.step(incoming_chats[0])
  File "/private/home/rebeccaqian/droidlet/base_agent/dialogue_manager.py", line 100, in step
    self.dialogue_stack.step()
  File "/private/home/rebeccaqian/droidlet/base_agent/dialogue_stack.py", line 54, in step
    output_chat, step_data = self.stack[-1].step()
  File "/private/home/rebeccaqian/droidlet/base_agent/dialogue_objects/interpreter.py", line 103, in step
    r = self.action_handlers[action_type](self.speaker, action_def)
  File "/private/home/rebeccaqian/droidlet/craftassist/agent/dialogue_objects/mc_interpreter.py", line 520, in handle_dance
    return new_tasks(), None, None
  File "/private/home/rebeccaqian/droidlet/craftassist/agent/dialogue_objects/mc_interpreter.py", line 493, in new_tasks
    dance_memid = random.choice(dance_memids)
  File "/private/home/rebeccaqian/.conda/envs/droidlet_env_clone/lib/python3.7/random.py", line 261, in choice
    raise IndexError('Cannot choose from an empty sequence') from None
IndexError: Cannot choose from an empty sequence
Traceback (most recent call last):
  File "craftassist_agent.py", line 332, in <module>
    sa.start()
  File "/private/home/rebeccaqian/droidlet/base_agent/core.py", line 20, in start
    self.handle_exception(e)
  File "/private/home/rebeccaqian/droidlet/base_agent/loco_mc_agent.py", line 211, in handle_exception
    raise e
  File "/private/home/rebeccaqian/droidlet/base_agent/core.py", line 18, in start
    self.step()
  File "/private/home/rebeccaqian/droidlet/base_agent/loco_mc_agent.py", line 216, in step
    super().step()
  File "/private/home/rebeccaqian/droidlet/base_agent/core.py", line 26, in step
    self.controller_step()
  File "/private/home/rebeccaqian/droidlet/base_agent/loco_mc_agent.py", line 282, in controller_step
    self.dialogue_manager.step(incoming_chats[0])
  File "/private/home/rebeccaqian/droidlet/base_agent/dialogue_manager.py", line 100, in step
    self.dialogue_stack.step()
  File "/private/home/rebeccaqian/droidlet/base_agent/dialogue_stack.py", line 54, in step
    output_chat, step_data = self.stack[-1].step()
  File "/private/home/rebeccaqian/droidlet/base_agent/dialogue_objects/interpreter.py", line 103, in step
    r = self.action_handlers[action_type](self.speaker, action_def)
  File "/private/home/rebeccaqian/droidlet/craftassist/agent/dialogue_objects/mc_interpreter.py", line 520, in handle_dance
    return new_tasks(), None, None
  File "/private/home/rebeccaqian/droidlet/craftassist/agent/dialogue_objects/mc_interpreter.py", line 493, in new_tasks
    dance_memid = random.choice(dance_memids)
  File "/private/home/rebeccaqian/.conda/envs/droidlet_env_clone/lib/python3.7/random.py", line 261, in choice
    raise IndexError('Cannot choose from an empty sequence') from None
IndexError: Cannot choose from an empty sequence
```

### PR DESCRIPTION
**Pull request by kavyasrinet**
# Description

This PR handles out  of scope dances

Fixes #341 

## Type of change

Please check the options that are relevant.

- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] Proposes a change (non-breaking change that isn't necessarily a bug)
- [ ] Refactor
- [ ] New feature (non-breaking change that adds a new functionality)
- [ ] Breaking change (fix or feature that would break some existing functionality downstream)
- [ ] This is a unit test
- [ ] Documentation only change
- [ ] Datasets Release
- [ ] Models Release

## Type of requested review

- [ ] I want a thorough review of the implementation.
- [x] I want a high level review. 
- [ ] I want a deep design review.

## Before and After

If this PR introduces a change or fixes a bug, please add before and after screenshots (if this is causes a UX change) or before and after examples(this could be plain text, videos etc ) to demonstrate the change.

If this PR introduces a new feature, please also add examples or screenshots demonstrating the feature.

# Testing

Tested with repro'ing the command locally

# Checklist:

- [x] I have performed manual end-to-end testing of the feature in my environment.
- [ ] I have added Docstrings and comments to the code.
- [ ] I have made changes to existing documentation where needed.
- [ ] I have added tests that show that the PR is functional.
- [ ] New and existing unit tests pass locally with my changes.
- [x] I have added relevant collaborators to review the PR before merge.

### CODE REVIEW

File: craftassist/agent/dialogue_objects/mc_interpreter.py [hunk 1]
```diff
@@ -485,14 +485,19 @@ def new_tasks():
...
+                    dance_mem = self.memory.get_mem_by_id(dance_memid)
+                    for i in range(repeat):
+                        dance_obj = dance.Movement(
+                            agent=self.agent, move_fn=dance_mem.dance_fn, dance_location=dance_location
+                        )
+                        t = self.task_objects["dance"](self.agent, {"movement": dance_obj})
+                        tasks_to_do.append(t)
+                else:
+                    # dance out of scope
+                    self.dialogue_stack.append_new(Say, "I don't know how to do that movement yet.")
```
Comment (aszlam) [inline craftassist/agent/dialogue_objects/mc_interpreter.py:499]: this should be ErrorWithRepsonse, no?

Comment (aszlam) [reply craftassist/agent/dialogue_objects/mc_interpreter.py:499]: still its an ErrorWithResponse as we've been using it (like if the agent doesn't find a reference object matching a tag, etc.)

Comment (aszlam) [reply craftassist/agent/dialogue_objects/mc_interpreter.py:499]: lol yeah, that is all the way back to Demi I think.  more generally we need to rewrite the hole detector, its insane rn.  but kind of getting derailed.

**Comment by kavyasrinet** [comment]
@aszlam Made the change, could you re-review please ?

**Review by aszlam — APPROVED** [review]

**Pull request merged**